src/avt/DBAtts/CMakeLists.txt

```diff
@@ -16,13 +16,17 @@
 #
 #****************************************************************************
 
-add_library(avtdbatts)
-
 add_subdirectory(MetaData)
 add_subdirectory(SIL)
 
-target_link_libraries(avtdbatts visitcommon)
-target_include_directories(avtdbatts PUBLIC ${VISIT_COMMON_INCLUDES})
+blt_add_library(
+    NAME       avtdbatts
+    SOURCES    ${avtdbatts_SOURCES}
+    INCLUDES   ${avtdbatts_INCLUDES}
+    DEPENDS_ON visitcommon)
+
+unset(avtdbatts_SOURCES CACHE)
+unset(avtdbatts_INCLUDES CACHE)
```

Comment:
Ok, so this maybe answers my prev. question. Is this the basic pattern...on var for headers and a companion var for sources both of which need to be unset after the `blt_add_library`?

Reply:
Yes, if visit_append_list is called, then unset for the created lists must be called as well, otherwise the Cache gets polluted, but also multiple calls to cmake-configure will duplicate items in the lists.

Reply:
Well, *pollution* sounds bad. Any way to craft things such that when CMake'ing is nearly complete (e.g. at the bottom of top-level `CMakeLists.txt`, some operations are performed to detect that *pollution* has happened and issue a warning message of some kind...e.g. `there appears to be missing unset() calls for <list-name>`

Reply:
There isn't a way (beyond manual parsing of the Cache) for the root CMakeLists.txt to know what vars have been created.

The need for 'unset' is indeed a possibly bad thing, but I could not figure a way around it except not use blt, or move all code out of subdirs  and have one monolithic CMakeLists in the parent dirs again (like we used to).

Reply:
I wasn't sure I totally understood here either but is `unset` in fact *required* or just a nice thing to do. I mean, do we really care if a bunch of variables that list source or include files get left around? If we adopt a standard naming convention, which you demonstrate already in the PR, there isn't really any possibility of a collision is there?

Reply:
The bigger problem is that we are 'appending' to a list, so if cmake-configure is called multiple times, without the 'unset' it will continually add to the list, making duplicates.
Would having the 'unset' at the beginning of the CMakeLists.txt be better or make things clearer? It would leave things in the cache, but still prevent the duplication issue.

I am going to test to see if there is another way, I can see this will be problematic.

Reply:
Ah, ok...so across re-runs of `cmake`, the problem would arise and compound. But, you mention duplicates.

If moving the `unset`s to the beginning means moving them further away from the `blt_add_library` call, maybe to another `CMakeLists.txt`, that would probably be even harder to maintain.

I dunno what this would mean performance wise but what about checking a list for dups upon an append operation and upon first encountering of a dup, treat it like its *starting over* and at that time, clear it before proceeding with the append?

Reply:
Ah, then how about a `visit_blt_add_library` that just does that in a combined operation?

Reply:
@markcmiller86 Thanks for the suggestion. I implemented `visit_add_library`, which calls `blt_add_library` and also makes the `unset` calls for the cache vars.  It doesn't matter if the vars don't exist, so it unsets all that may have been created.  The list may need to be updated as I go through the other libraries and update them for BLT.
What do you think of the change?

Reply:
I think adding `visit_add_library` that calls both the `blt_add_library` and `unset` functions is a good idea. That makes it easier to prevent mistakes in the future.